### ISSUE TITLE
Cannot submit Analytics settings

### ISSUE THREAD

**Issue by eclarke1** [google/site-kit-wp#3535]
## Bug Description

After I set up the Analytics module with both my UA and GA4 properties, they show up in the Analytics settings panel as expected, in both the "view" and "edit" state.

Problem: However, when I edit settings, it does not let me submit. In the attached screenshot for example, I just changed the tracking exclusion setting from "All logged-in users" to selecting none, but the submit button is still disabled, although I've made a change. There is also nothing else that would make me think I cannot submit here.

It looks like this is related to the GA4 property: Once I change that blue notice dropdown to e.g. "Set up a new property", it allows me to submit.

[Related Asana task](https://app.asana.com/0/1200316599642381/1200443654792717)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The GA4 integration should not prevent the user from submitting the Analytics settings panel after changing a setting - it shouldn't be required to first change (then change back) the GA4 property in the dropdown.
* The setting and validating of the web data stream for the selected GA4 property should work as expected:
    * If there is a matching web data stream in the property for the current site, that one should automatically be set.
    * If there is no matching web data stream in the property for the current site, it should still be possible to submit the form; in that case the web data stream needs to be created on submission, and then set in the module (with both web data stream ID and measurement ID).

## Implementation Brief

* Merge #3565

### Test Coverage

* N/A

### Visual Regression Changes

* N/A

## QA Brief

* Activate and connect the Analytics module. 
* Go to the Analytics settings, make changes and verify that you can save it.

## Changelog entry

* N/A

**Comment by eugene-manuilov** [comment]
@eclarke1 @aaemnnosttv a fix for #3538 should fix this issue as well.

**Comment by wpdarren** [comment]
### QA Update: Fail ❌ 

@eugene-manuilov when `ga4setup` is enabled on the tester plugin, I am unable to save any Analytics settings. 

Here's a screencast:

https://user-images.githubusercontent.com/73545194/122088296-7fd4bb00-cdfd-11eb-9d5f-868114d8de43.mp4

**Comment by eugene-manuilov** [comment]
@wpdarren ok, this ticket also depends on #3571, once it's merged - your issue should be fixed.

**Comment by aaemnnosttv** [comment]
#3571 has been merged; sending back to @wpdarren for another pass 👍

**Comment by aaemnnosttv** [comment]
@eugene-manuilov – can you cherry pick the changes from #3571 into a new PR for main? I didn't realize that this was needed here as QA for this issue is blocked by it.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv could you please review it? #3587

**Comment by aaemnnosttv** [comment]
Merged! Sending to @wpdarren for another pass 👍

**Comment by wpdarren** [comment]
### QA Update: Pass ✅ 

- On the Analytics settings, I am able to make changes and save them.  
- The `confirm changes` button is active when a setting is changed. 
- When I go back into the settings, the changes have been saved and are updated accordingly. 

![image](https://user-images.githubusercontent.com/73545194/122213599-07273a80-cea1-11eb-8c98-5754dd53f399.png)
